Assets/Scripts/Models/Trade/Currency.cs

```diff
@@ -8,13 +8,31 @@
 #endregion
 
 using System.Xml;
+using System;
 
 public class Currency
 {
     public string Name;
     public string ShortName;
+    
     public float Balance;
     
+    public void SetBalance (float value) {
```

Comment:
I agree, this would be better as a get/set property with banlanceChanged called on the set when the value differ.